service/history/handler.go

```diff
@@ -202,15 +202,18 @@ func (h *Handler) DeepHealthCheck(
 		return nil, err
 	}
 	if status.Status != healthpb.HealthCheckResponse_SERVING {
+		metrics.HistoryHostHealthGauge.With(h.metricsHandler).Record(float64(enumsspb.HEALTH_STATE_DECLINED_SERVING))
```

Comment:
is this metrics per host?

Reply:
yes! Because this is on the handler, metrics get scraped by prometheus on per-cell and per-host information. We'll be able to create a graph with this on per-cluster per-host checks